src/Sylius/Bundle/AdminBundle/Command/CreateAdminUserCommand.php

```diff
@@ -0,0 +1,112 @@
+    description: 'Create a new admin user'
+)]
+final class CreateAdminUserCommand extends Command
+{
+    private SymfonyStyle $io;
+
+    public function __construct(
+        private UserRepositoryInterface $adminUserRepository,
+        private FactoryInterface $adminUserFactory,
+    ) {
+        parent::__construct();
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output): void
+    {
+        $this->io = new SymfonyStyle($input, $output);
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        if (!$input->isInteractive()) {
+            $this->io->error('This command should be run interactively.');
+
+            return Command::FAILURE;
+        }
+
+        $this->io->title('Admin user creation');
+
+        $confirm = $this->io->confirm('Do you want to create an admin user ?', true);
+
+        if (!$confirm) {
+            return Command::INVALID;
+        }
+
+        $email = $this->io->askQuestion($this->createEmailQuestion());
+
+        if ($this->checkIfAdminUserExists($email)) {
+            $this->io->error(sprintf('Admin user with email address %s already exists.', $email));
+
+            return COMMAND::FAILURE;
+        }
+
+        $userName = $this->io->ask('Username');
+        $firstName = $this->io->ask('Firstname');
+        $lastName = $this->io->ask('Lastname');
+        $password = $this->io->askHidden('Password');
+
+        /** @var AdminUserInterface $adminUser */
+        $adminUser = $this->adminUserFactory->createNew();
+
+        $adminUser->setEmail($email);
+        $adminUser->setPlainPassword($password);
+        $adminUser->setUsername($userName);
+        $adminUser->setFirstName($firstName);
+        $adminUser->setLastName($lastName);
+
+        $locales = Locales::getNames();
+
+        $localeCode = $this->io->choice('Select the locale code', $locales, 'en_US');
+        $adminUser->setLocaleCode($localeCode);
+
+        $enabled = $this->io->confirm('Do you want to enabled this admin user?', true);
+        $adminUser->setEnabled($enabled);
+
+        $this->adminUserRepository->add($adminUser);
+
+        $this->io->success(sprintf('Admin user %s was successfully created', $adminUser->getEmail()));
+
+        return Command::SUCCESS;
+    }
+
+    private function createEmailQuestion(): Question
+    {
+        $question = new Question('Email');
+        $question->setValidator(function (string $email) {
+            if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
+                $this->io->error('Email address is not valid. Please, enter a valid address');
+
+                return false;
+            }
+
+            return $email;
+        });
+        $question->setMaxAttempts(3);
+
+        return $question;
+    }
+
+    private function checkIfAdminUserExists(string $email): bool
+    {
+        return null !== $this->adminUserRepository->findOneByEmail($email);
```

Comment:
We should check it by canonicalized email

Reply:
I think that this is fine as the custom `findOneByEmail` method actually checks by the canonical email

Reply:
Yes, this method checks by field `emailCanonical`, which is correct, but the email value passed to this repository method is not canonicalized, so there is a possibility that this check will not work properly.